packages/browser-wallet/src/background/update.ts

```diff
@@ -0,0 +1,67 @@
+import { StorageAccessor, storedCredentials, storedIdentities } from '@shared/storage/access';
+import { Identity, WalletCredential } from '@shared/storage/types';
+
+const identityLock = 'Concordium_identity_lock';
+const credentialLock = 'Concordium_credential_lock';
+
+/**
+ * Generic method to add an element to list in storage
+ */
+async function addToList<Type>(lock: string, addition: Type, storage: StorageAccessor<Type[]>): Promise<void> {
+    return navigator.locks.request(lock, async () => {
+        const list = (await storage.get()) || [];
+        return storage.set(list.concat(addition));
+    });
+}
+
+/**
+ * Generic method to edit/update elements in a list in storage
+ * Note that this replaces the element found by the findPredicate with the edit.
+ */
+async function editList<Type>(
+    lock: string,
+    edits: Type[],
+    findPredicate: (current: Type) => (candidate: Type) => boolean,
+    storage: StorageAccessor<Type[]>
+): Promise<void> {
+    return navigator.locks.request(lock, async () => {
+        const currentList = await storage.get();
+        if (!currentList) {
+            return;
```

Comment:
It may be a better idea to throw an error in this case. I am trying to edit a non-existing list and the method just goes well, so I am now under the assumption that my edit went through, but nothing happened.